hw2/01-coins.js

```diff
@@ -1,7 +1,84 @@
+/**
+ * Naya Mairena
+ * Fall 2022
+ * CS465P Full Stack
+ */
 /** Exercise 01 - Coins **/
 
+//Function that will use a conditional (ternary) operator to check if the change name needs to be plural.
+const plurals = (count, word) => {
+  //Check if last letter of the string ends with a "y". This is because "penny" is different than the others.
+  if (word.slice(-1) === 'y') {
+    //Slice "penny" to be "penn".
+    let newWord = word.slice(0, -1);
+    //Return "pennies" if there is more than 1 penny.
+    return count === 1 ? word : newWord + 'ies';
+  } else return count === 1 ? word : word + 's';
+  //Else, we want to return the word with an "s" at the end for multiples.
+};
+
+//Function that will take a dollar amount as input and return its change broken down by coins.
 const calculateChange = (input) => {
   // Add your code here
+  //Multiple use cases to check before actual functionality.
+  if (input > 10) return 'Error: the number is too large.';
+  else if (input === 0) return 'Error: 0 returns no change.';
+  else if (input < 0) return 'Error: Negative dollar amount not allowed.';
+  else if (isNaN(input)) return 'Error: Input is not a number value.';
+  else {
+    //I decided to fix the values to be up to two decimal places.
+    //I was getting the wrong change amounts with regular float values.
+    let change = input.toFixed(2);
+    //Get the number of dollars by rounding down.
+    let dollars = Math.floor(input);
+    //Calculate a new value for the remaining change.
+    change = (change % 1).toFixed(2);
+    let quarters = Math.floor(change / 0.25);
+    change = (change % 0.25).toFixed(2);
+    let dimes = Math.floor(change / 0.1);
+    change = (change % 0.1).toFixed(2);
+    let nickels = Math.floor(change / 0.05);
+    change = (change % 0.05).toFixed(2);
+    let pennies = Math.floor(change / 0.01);
```

Comment:
This is an interesting implementation. it took me a second to realize what was going on and I got to see the toFixed method in action.